satellite/gracefulexit/endpoint.go

```diff
@@ -709,7 +711,7 @@ func (endpoint *Endpoint) getFinishedMessage(ctx context.Context, signer signing
 	return message, nil
 }
 
-func (endpoint *Endpoint) updatePointer(ctx context.Context, exitingNodeID storj.NodeID, receivingNodeID storj.NodeID, path []byte, pieceNum int32) (err error) {
+func (endpoint *Endpoint) updatePointer(ctx context.Context, originalPointer *pb.Pointer, exitingNodeID storj.NodeID, receivingNodeID storj.NodeID, path []byte, pieceNum int32) (err error) {
 	defer mon.Task()(&ctx)(&err)
 
 	// remove the node from the pointer
```

Comment:
If the pointer can change during graceful exit, then we shouldn't use the new pointer to find the piece we need to remove right since it could be a different piece from what we asked the node to transfer?

Reply:
I thought the same thing in my original code change.  But the 20 node, 1000 objects, 5 exiting test failed miserably due to a node having more than 1 piece for a segment.  My theory is another exiting node transferred a piece to the now replacement node.

Reply:
If the pointer has changed from the `originalPointer`, it should fail - so I don't understand how the receiving node could ever be in the pointer and `updatePointer` would still succeed.